### PR TITLE
Moving GET and POST endpoints from /users to /api/users

### LINKED ISSUE #2: Move /users endpoints from /src/index.ts to /src/api/users.ts

### PR DESCRIPTION
**Pull request by dyazdani**
## Description
Moved GET and POST users endpoints from `index.ts` file to a `users.ts` file and switched the endpoint from /users to /api/users. These endpoints are primarily for testing at this point, but POSTing a user will eventually involve registering at a /api/users/register endpoint.

## Related Issue
Closes #2 

## Checklist
 - [x] Created tests which fail without the change (if possible)
 - [x] Confirmed all tests are passing
 
## How Has This Been Tested?
Postman

Screen Shot 2023-10-17 at 10 07 34
Screen Shot 2023-10-17 at 10 08 41

### CODE REVIEW

File: src/api/users.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                email,
+                username,
+                password
+            }
+        })
+        res.send({user});
+    } catch (e) {
+        next(e)
+    }
+})
```
Comment (paloobi) [inline src/api/users.ts:35]: Remember to delete this once you add register - might be good to add a TODO to that effect

File: src/api/users.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    try {
+        const users = await prisma.user.findMany();
+        res.send({users});
+    } catch (e) {
+        next(e);
+    }
+})
+
+// POST /api/users
+
```
Comment (paloobi) [inline src/api/users.ts:20]: nit: delete newline?
```suggestion
```

File: src/api/users.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+const usersRouter = express.Router();
+
+// GET /api/users
+
+usersRouter.get("/", async (req, res, next): Promise<void> => {
```
Comment (paloobi) [inline src/api/users.ts:10]: You may want to add a TODO here of some sort to restrict or delete this endpoint. eg. maybe only admins can see all users, or maybe only logged in users?

File: src/api/users.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    try {
+        const users = await prisma.user.findMany();
```
Comment (paloobi) [inline src/api/users.ts:12]: Also add a TODO to strip the passwords from these users - it would be a security issue if you forget to do that later

**Review by paloobi — APPROVED** [review]

File: src/api/users.ts (hunk 3 above)
Comment (dyazdani) [reply src/api/users.ts:10]: I'd be fine deleting it.

**Pull request merged**